crates/lib/src/identity.rs

```diff
@@ -101,16 +107,25 @@ impl hex::FromHex for Identity {
     }
 }
 
+impl FromStr for Identity {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Self::from_hex(s).context("Invalid Identity hex string")
+    }
+}
+
 #[cfg(feature = "serde")]
 impl serde::Serialize for Identity {
     fn serialize<S: serde::Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
-        spacetimedb_sats::ser::serde::serialize_to(self, serializer)
+        spacetimedb_sats::ser::serde::serialize_to(self.as_bytes(), serializer)
```

Comment:
This is a breaking change -- are we sure consumers can handle it?

Reply:
AFAICT, there's nowhere that actually serializes it to serde directly, and it would be a pretty useless serialization anyway. At the time I was citing the prior art of Address having a different serde repr than its SATS repr, but now it seems it's been split into AddressForUrl. In any case - I believe that there's not actually a problem with this being distinct, as the SATS repr and serde repr are used in very different contexts, and if it's being serialized to JSON using the sats::serde::serialize_to it'll still be fine.

Reply:
`Identity` seems to be used in `RecoveryCodeResponse`at least, which is a plain JSON object. It will look funny to have a byte array in there, and it will break older clients talking to a newer server. If you think the space saving is worth it, maybe branch on `is_human_readable`?

Reply:
Oh, no that's actually what it does already - serialize_to branches on is_human_readable and makes it hex if it is

Reply:
Oh right, we put a serde in the serde